src/routing_table/network_tests.rs

```diff
@@ -166,15 +170,18 @@ impl Network {
                             Network::store_merge_info(&mut merge_other_info,
                                                       *target_node.our_group_prefix(),
                                                       (targets, merge_details));
-                            // add needed contacts
-                            let needed = target_node.needed().clone();
-                            for needed_contact in &needed {
-                                let _ = target_node.add(*needed_contact);
+                            // Forcibly add new connections.
+                            for name in node_expected.iter() {
+                                // Try adding each node we should be connected to.
+                                // Ignore failures and ignore splits.
+                                let _ = target_node.add(*name);
```

Comment:
Should `name ` be removed from `node_expected`, except if `add` returned an error?